dev/com.ibm.ws.messaging.comms.server/src/com/ibm/ws/jfap/inbound/channel/CommsServerServiceFacade.java

```diff
@@ -77,7 +90,7 @@ public class CommsServerServiceFacade {
     /** Lock to guard chain actions (update,stop and sslOnlyStop).. as of now as all chain actions are executed by SCR thread */
     private final Object actionLock = new Object();
```

Comment:
We now have two locks    (the actionLock and the synchronised methods) so we had better to be careful to always lock them in the same order through all code paths that can use both simultaneously :-)

Reply:
In this case, if we have any synchronised method that locks the actionLock, then we can't enter our 'first' synchronised method while already holding the action lock